### DISCUSSION TITLE
Add php docs

### DISCUSSION THREAD

**Discussion by eerison** [LunarVim/LunarVim#3546] [Ideas]
I see that many guys use lunarvim with php, But I dont see too much docs into the site related with it :/ !

**Comment by bricebou** [comment]
Hello,

I'm a complete noob discovering (Lunar)Vim and I'm trying to write some PHP with it.
I've managed to launch the Intelephense LSP module (with a needed composer.json) but I have no idea on how to use its functionalities, as generating getters ans setters, adding PHP DocBlock...

Is there anyone around who can help me understand how to set an efficient programming environment for PHP, including phpcs, phpcbf etc.?

Thanks in advance! :)

**Comment by eerison** [comment]
I didn't finish my environment yet, but as I can see phpactor is recommended for PHP environment 

https://github.com/phpactor/phpactor

Maybe it can help you get some start 

https://github.com/shield-wall/lunarvim-docker/blob/master/Dockerfile

**Reply by bricebou** [reply]
Thanks a lot, I'll give it a try !

**Reply by MatheusPMelo** [reply]
Hey guys! I start my travel in lunar and i amazing this expirence. I recomend you use intelephense because you can set some settings with command :LspSettings intelephense. This command go create a json file for you type your custom settings. I'm reading for this now.

sorry my english, i'm brazilian ksksks bye bye

**Reply by MatheusPMelo** [reply]
Dont forget close your question ;)
